lib/utils/addEntries.js

```diff
@@ -143,43 +158,31 @@ function addEntries(compiler, options, server) {
       additionalEntries.push(hotEntry);
     }
 
-    // use a plugin to add entries if available
-    // @ts-ignore
-    if (webpack.EntryPlugin) {
-      // use existing plugin if possible
-      let entryPlugin = config.plugins.find(
-        (plugin) => plugin.constructor.name === 'DevServerEntryPlugin'
+    // use a hook to add entries if available
+    if (EntryPlugin) {
+      compiler.hooks.make.tapPromise('DevServerPlugin', (compilation) =>
+        Promise.all(
+          additionalEntries.map(
+            (entry) =>
+              new Promise((resolve, reject) => {
+                compilation.addEntry(
+                  compiler.context,
+                  EntryPlugin.createDependency(entry, {}),
+                  {}, // global entry
+                  (err) => {
+                    if (err) return reject(err);
+                    resolve();
+                  }
+                );
+              })
+          )
+        )
       );
-
-      if (entryPlugin) {
-        entryPlugin.entries = additionalEntries;
-      } else {
-        entryPlugin = new DevServerEntryPlugin(
-          compiler.context,
-          additionalEntries,
-          {} // global entry
-        );
-        config.plugins.push(entryPlugin);
-        entryPlugin.apply(compiler);
-      }
     } else {
       config.entry = prependEntry(config.entry || './src', additionalEntries);
       compiler.hooks.entryOption.call(config.context, config.entry);
     }
-
-    if (options.hot || options.hot === 'only') {
-      config.plugins = config.plugins || [];
-      if (
-        !config.plugins.find(
-          // Check for the name rather than the constructor reference in case
-          // there are multiple copies of webpack installed
-          (plugin) => plugin.constructor.name === 'HotModuleReplacementPlugin'
-        )
```

Comment:
This is changed to constructor check. I wonder the statement is valid, as multiple webpack usually leads to error.

Reply:
Yep, good catch